### PR TITLE
doc: add terminal program configuration

### PR DESCRIPTION
**Pull request by OlegHahm**
### Contribution description

This is mostly an import from https://github.com/RIOT-OS/RIOT/wiki/Configure-terminal-programs-for-correct-display-of-newlines.

### Testing procedure

Checkout this branch, run `make doc` and check the link "Terminal programs configuration" in the navigation menu.

### CODE REVIEW

File: doc/doxygen/src/terminal-programs.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+Some terminals need more, for example, a carriage return and a line feed
+character (0xD, 0xA). See https://en.wikipedia.org/wiki/Newline for background.
+
+This page tries to collect needed settings for common terminal programs that
+will make them correctly display newlines.
+
+picocom                                                               {#picocom}
+=======
+Start with `--imap lfcrlf` parameter.
```
Comment (benpicco) [inline doc/doxygen/src/terminal-programs.md:21]: You can also call `make term RIOT_TERMINAL=picocom`

**Review by benpicco — COMMENTED** [review]
We actually have build-in support for serveral terminal programs in `makefiles/tools/serial.inc.mk`

File: doc/doxygen/src/terminal-programs.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+- Interactive method:
+    - Press ***Ctrl+A u***.
+- Manual method:
+    - Edit the configuration file (`~/.minirc.dfl` per default).
+    - Add the following line:
+    pu addcarreturn     Yes
+
+miniterm                                                             {#miniterm}
+========
+Start with `--eol CR`parameter.
```
Comment (benpicco) [inline doc/doxygen/src/terminal-programs.md:43]: Or call `make term RIOT_TERMINAL=miniterm`

**Review by benpicco — APPROVED** [review]

**Review by maribu — APPROVED** [review]
Thx :) Please squash

**Pull request merged**